### PR TITLE
feat(doc2vec) : Semantic Text Similarity Algorithm with dataset & training code

### PR DESCRIPTION
**Pull request by hastagAB**
# Description

New Open Source License Scanning Algorithm: **Semantic Text Similarity** find similarity between documents according to its semantics. 
The [Gensim](https://radimrehurek.com/gensim/models/doc2vec.html) implementation of [Doc2Vec](https://cs.stanford.edu/~quocle/paragraph_vector.pdf) converts the whole document (unlike word2vec) into vector with their labels.
The Doc2Vec model is trained using the filename as its label and license text as the document. 
The current training dataset is the txt format of  [license-list-data](https://github.com/spdx/license-list-data) provided by SPDX.

## Files
- semanticTextSim.py (Implementation of Algorithm)
- spdxDoc2Vec.model (the Trained model in binary)
- train.py (Code to train the model)
- text (Folder containing all the spdx license dataset)

## Test 

- ### Test the agent for scanning any file for license statements
` atarashi -a semanticTextSim <pathToFile> `
Currently, it returns the license name with the highest Cosine Sim Score.
> Note: The agent is able to return top ten most similar license names with its sim score

- ### Train the model (Optional)
cd to ```semanticTextSim``` folder.
Run Command: `python train.py`

### CODE REVIEW

**Comment by amanjain97** [comment]
Please fix the travis CI. Code looks good. 
Nice work. I think the accuracy of the model is not good but we can improve later.
@hastagAB

File: setup.py [hunk 1]
```diff
@@ -68,6 +69,7 @@ def read(fname):
   'pyxDamerauLevenshtein>=1.5',
   'code_comment@git+https://github.com/amanjain97/code_comment@master#egg=code_comment',
   'urllib3>=1.24.1'
```
Comment (GMishx) [inline setup.py]: Please insert the comma `,`

**Comment by hastagAB** [comment]
Have rebasing issue in branch. Unable to fix.
Closing this
